src/shinylive.ts

```diff
@@ -542,12 +542,10 @@ function shinyliveUrlDecode(url: string): ShinyliveBundle | undefined {
   });
 
   const pathParts = pathname.split("/");
+  const language: ShinyliveLanguage = pathParts.includes("py") ? "py" : "r";
+  const mode: ShinyliveMode = pathParts.includes("editor") ? "editor" : "app";
 
-  return {
-    language: pathParts[1] as ShinyliveLanguage,
-    files: files as ShinyliveFile[],
-    mode: pathParts[2] as ShinyliveMode,
-  };
+  return { language, mode, files: files as ShinyliveFile[] };
```

Comment:
Okay, technically this is an unrelated change and normally I'd do it in a separate PR. But this is small, easy, and actually has no impact other than making sure I don't forget. (We don't actually use `language` or `mode`, we only need `files` downstream of this function.)